data_structures/hashing/hash_table.py

```diff
@@ -21,6 +21,29 @@ def __init__(
         self._keys: dict = {}
 
     def keys(self):
```

Comment:
I don't love this name `.keys()` because `{0: 5, 4: 4, 3: 3, 2: 2, 1: 1}.keys()` returns `dict_keys([0, 4, 3, 2, 1])` which is a `list`, not a `dict`.  But let's leave those considerations for a separate pull request.
* (`https://docs.python.org/3/library/stdtypes.html#object.__dict__`)[https://docs.python.org/3/library/stdtypes.html#object.__dict__]